### PR TITLE
Refactoring for more consistency

### PR DESCRIPTION
**Pull request by d10r**
After the merge of https://github.com/superfluid-finance/custom-supertokens/pull/10 (which I didn't review in time before merged) I noticed naming inconsistencies. E.g. the newly added token contract was named _MaticBridgedPureSuperTokenProxy_ while the other example tokens omit the _Pure_ (except the minimal one having no own distinct features) and the _Proxy_ (despite deriving from `UUPSProxy` too).  
Being onto it, I also did several other small changes which I think make sense.
In the Readme I tried to take the perspective of an outsider who's not yet super familiar with the concepts of pure (ex "native") vs wrapper tokens, or with proxies. Also added a sentence about what "custom Super Token" actually means.  
I also used the term "hybrid proxy" in order to name this composition of proxy + custom functionality. Not sure if adding such a term helps, but I feel it could evoke the right intuition about how this contracts work.

I think the addition of the factory initialization in the `_initialize` of the token contracts is a good idea, so I replicated that in `MaticBridgedSuperToken`.

Still to be done (imo):
- "fix" indentation: switch to spaces in order to be consistent with other repos
- "fix" code style: afaik we use JS with semicolons everywhere else, should imo stick to it
- be a bit more explicit about how to deploy. The `scripts/README.md` doesn't really tell. If guessing and running `yarn run <script>) (after figuring out there's npm scripts in `package.json`), one gets something like _something went wrong while attempting to connect to the network at http://127.0.0.1:7545_...
- add deploy script for the MaticBridgedSuperToken (I'd do that once I know how to test it)
- add a mention about deployment to the root README (maybe better keep just that one)
- (stretch goal) use https://github.com/superfluid-finance/metadata instead of a hardcoded list of deployments

### CODE REVIEW

File: truffle-config.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ module.exports = {
 	// Configure your compilers
 	compilers: {
 		solc: {
-			version: "^0.8.0"
+			version: "0.8.14"
```
Comment (d10r) [inline truffle-config.js:40]: fixed in order to avoid truffle to fetch a compiler list at every invocation

**Comment by hellwolf** [comment]
nice

**Review by JoshuaTrujillo15 — APPROVED** [review]
LGTM

**Comment by ngmachado** [comment]
Going to merge, even if @d10r is a impersonator that don't signed the PR.

**Pull request merged**